pkg/continuoustest/util.go

```diff
@@ -10,13 +10,89 @@ import (
 	"time"
 
 	"github.com/prometheus/common/model"
+	"github.com/prometheus/prometheus/model/histogram"
 	"github.com/prometheus/prometheus/prompb"
+	"github.com/prometheus/prometheus/storage/remote"
+
+	"github.com/grafana/mimir/pkg/mimirpb"
 )
 
 const (
 	maxComparisonDelta = 0.001
 )
 
+type generateHistogramFunc func(t time.Time) prompb.Histogram
+type generateSeriesFunc func(name string, t time.Time, numSeries int) []prompb.TimeSeries
+type generateValueFunc func(t time.Time) float64
+type generateSampleHistogramFunc func(t time.Time) *model.SampleHistogram
+
+var (
+	generateHistogram = []generateHistogramFunc{
```

Comment:
Seems like these arrays need to line up, so wouldn't it be nicer to define something like
```
type HistogramProfile {
  metricName string
  typeLabel      string
  generate generateHistogramFunc
  generateSeries generateSeriesFunc
  generateValue generateValueFunc
  generateSample generateSampleHistogramFunc
}
```

And fill it out for each of the 4 ? that way these are made sure to be in sync